lang_java/tree_sitter/tree_sitter_ast_java.ml

```diff
@@ -0,0 +1,707 @@
+  right: _type;
+}
+
+and integral_type = 
+| Integral of string wrap
+
+and interface_body = 
+  | Annotation of annotation_type_declaration list option 
+  | Class of class_declaration list option 
+  | Constant of constant_declaration list option 
+  | Enum of enum_declaration list option 
+  | Interface of interface_declaration list option 
+  | Method of method_declaration list option 
+
+and interface_declaration = {
+  body: interface_body;
+  name: identifier;
+  type_parameters: type_parameters option;
+  mods: extends_interfaces list option * modifiers list option;
+
+}
+
+and interface_type_list = 
+  | Type of _type list 
+
+and labeled_statement = 
+  | Statement of _statement list 
+  | Identifier of identifier list 
+
+and lambda_expression = {
+  body: _expression * block;
+  parameters: formal_parameters * identifier * inferred_parameters;
+}
+
+and local_variable_declaration = {
+  declarator: variable_declarator list;
+  dtype: _unannotated_type;
+  mods: modifiers option;
+
+}
+
+and marker_annotation = {
+  name: identifier * scoped_identifier;
+}
+
+and method_declaration = {
+  body: block option;
+  dimensions: dimensions option;
+  name: identifier;
+  parameters: formal_parameters;
+  dtype: _unannotated_type;
+  type_parameters: type_parameters option;
+  mods: annotation list option * marker_annotation list option * modifiers list option * throws list option;
+
+}
+
+and method_invocation = {
+  arguments: argument_list;
+  name: identifier;
+  dobject: _primary option * identifier option * scoped_identifier option * super option;
+  type_arguments: type_arguments option;
+  mods: super option;
+
+}
+
+and method_reference = 
+  | Primary of _primary list 
+  | Type of _type list 
+  | Identifier of identifier list 
+  | Scoped of scoped_identifier list 
+  | Super of super list 
+  | Type1 of type_arguments list 
+
+and modifiers = 
+  | Annotation of annotation list option 
+  | Marker of marker_annotation list option 
+
+and module_declaration = 
+  | Annotation of annotation list 
+  | Identifier of identifier list 
+  | Marker of marker_annotation list 
+  | Module of module_directive list 
+  | Scoped of scoped_identifier list 
+
+and module_directive = 
+  | Identifier of identifier list 
+  | Module of module_name list 
+  | Requires of requires_modifier list 
+  | Scoped of scoped_identifier list 
+
+and module_name = 
+  | Identifier of identifier list 
+  | Module of module_name list 
+
+and object_creation_expression = {
+  arguments: argument_list;
+  dtype: _simple_type;
+  type_arguments: type_arguments option;
+  mods: _primary list option * class_body list option * identifier list option * scoped_identifier list option;
+
+}
+
+and package_declaration = 
+  | Annotation of annotation list 
+  | Identifier of identifier list 
+  | Marker of marker_annotation list 
+  | Scoped of scoped_identifier list 
+
+and parenthesized_expression = 
+  | Expression of _expression 
+
+and program = 
```

Comment:
when a type has only one choice, you can actually remove the constructor name as it's mostly useless.
you can just use a type alias as in 
and program = _statement list option

See ast_java.ml for examples of that.